encord/storage_cloud_data_migration.py

```diff
@@ -0,0 +1,178 @@
+from encord import EncordUserClient
+from encord.http.bundle import BundlablePayload, Bundle, bundled_operation
+from encord.http.v2.api_client import ApiClient
+from encord.orm.base_dto import BaseDTO
+from encord.orm.cloud_integration import CloudIntegration
+from encord.storage import StorageItem
+
+STORAGE_CLOUD_DATA_MIGRATION_BUNDLE_LIMIT = 1000
+
+
+def update_storage_item_cloud_info(
+    user_client: EncordUserClient,
+    item: Union[StorageItem, UUID, str],
+    new_url: Optional[str] = None,
+    new_cloud_integration: Optional[Union[CloudIntegration, str, UUID]] = None,
+    from_cloud_integration: Optional[Union[CloudIntegration, str, UUID]] = None,
+    verify_access: bool = True,
+    skip_missing: bool = False,
+    bundle: Optional[Bundle] = None,
+) -> None:
+    """
+    Update the cloud storage information of a storage item. This can be used to avoid re-importing data when the cloud
+    configuration changes (and thus the data is accessible through a different cloud integration), or if the data has
+    been moved to a different location (and thus the URL has changed).
+
+    :param user_client: The user client to use.
+    :param item: The item to update. Can be either a StorageItem instance, the item's UUID, or the URL of the item.
+    :param new_url: The new URL to set. URL will be left unchanged if `None` is passed.
+    :param new_cloud_integration: The new cloud integration to set. Cloud integration will be left unchanged if `None`
+        is passed.
+    :param from_cloud_integration: The cloud integration to update from. Acts as a check: no update will be performed if
+        cloud integration of the item does not match this value. If `None`, the check will be skipped.
+    :param verify_access: Whether to verify access to the item.
+    :param skip_missing: if true, no error will be raised if the item is not found. Otherwise, the operation (and
+        the operations in the same bundle) will be cancelled and an error will be raised.
+    :param bundle: The optional :class:`encord.http.bundle.Bundle` instance used to group updates into bulk calls.
+    """
+    if not bundle:
+        with Bundle() as bundle:
+            update_storage_item_cloud_info(
+                user_client=user_client,
+                item=item,
+                new_url=new_url,
+                new_cloud_integration=new_cloud_integration,
+                from_cloud_integration=from_cloud_integration,
+                verify_access=verify_access,
+                skip_missing=skip_missing,
+                bundle=bundle,
+            )
+            return
+
+    item_uuid: Optional[UUID] = None
+    item_url: Optional[str] = None
+    if isinstance(item, StorageItem):
+        item_uuid = item.uuid
+    elif isinstance(item, UUID):
+        item_uuid = item
+    else:  # it's a string
+        try:
+            item_uuid = UUID(item)
+        except ValueError:
+            item_url = item
+
+    if not item_uuid and not item_url:
```

Comment:
Can you even hit this condition? Looks like we'll hit item_url = item (Assuming that attempt UUID(x) always errors with ValueError

Reply:
This is mostly to be obvious enough for `mypy` to figure it out.